src/Microsoft.Health.Core/Features/HealthPublisher/HealthCheckPublisher.cs

```diff
@@ -0,0 +1,30 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Health.Core.Features.Health;
+
+internal class HealthCheckPublisher : IHealthCheckPublisher
```

Comment:
Maybe something like `InMemoryHealthCheckPublisher`?

Reply:
I changed it to `HealthCheckCachePublisher` to be more similar to the other naming - `ValueCache`, `CachedHealthCheckMiddleware`, etc.